Nodejs/Product/Nodejs/SharedProject/ProjectNode.cs

```diff
@@ -6511,15 +6511,24 @@ public virtual int ParseCanonicalName(string name, out uint itemId)
             // we always start at the current node and go it's children down, so 
             //  if you want to scan the whole tree, better call 
             // the root
-            name = EnsureRootedPath(name);
-            itemId = 0;
-            var child = FindNodeByFullPath(name);
-            if (child != null)
+            try
             {
-                itemId = child.HierarchyId;
-                return VSConstants.S_OK;
+                name = EnsureRootedPath(name);
+                var child = FindNodeByFullPath(name);
+                if (child != null)
+                {
+                    itemId = child.HierarchyId;
+                    return VSConstants.S_OK;
+                }
+            }
+            catch (ArgumentException)
+            {
+                // This is expected when the path contains 
```

Comment:
Doc's say ArgumentException is the only one, since the method doesn't do any actual IO.